packages/@angular/cli/tasks/render-universal.ts

```diff
@@ -0,0 +1,33 @@
+import { requireProjectModule } from '../utilities/require-project-module';
+import { join } from 'path';
+
+const fs = require('fs');
+const Task = require('../ember-cli/lib/models/task');
+
+export interface RenderUniversalTaskOptions {
+  inputIndexPath: string;
+  route: string;
+  serverOutDir: string;
+  outputIndexPath: string;
+}
+
+export default Task.extend({
+  run: function(options: RenderUniversalTaskOptions): Promise<any> {
+    require('zone.js/dist/zone-node');
+
+    const renderModuleFactory =
+      requireProjectModule(this.project.root, '@angular/platform-server').renderModuleFactory;
+
+    // Get the main bundle from the server build's output directory.
+    const serverDir = fs.readdirSync(options.serverOutDir);
+    const serverMainBundle = serverDir
+      .filter((file: string) => /main\.[a-zA-Z0-9]{20}.bundle\.js/.test(file))[0];
```

Comment:
What happens if the user asks for the app to be built without hashes? Since we propagate those options to the server build, the server main file won't match this pattern, I don't believe.

Reply:
I will test this, if it fails, I will tweak the regex.